src/include/network/widgets.rb

```diff
@@ -559,46 +559,45 @@ def handleDevice(items, selected)
     # @param [Array<Fixnum>] itemIds           list of indexes into LanItems::Items
     # @param [Array<String>] enslavedIfaces    list of device names of already enslaved devices
     def CreateSlaveItems(itemIds, enslavedIfaces)
-      itemIds = deep_copy(itemIds)
-      enslavedIfaces = deep_copy(enslavedIfaces)
+      raise ArgumentError if itemIds.nil?
+
       items = []
 
-      Builtins.foreach(itemIds) do |itemId|
-        description = ""
+      itemIds.each do |itemId|
         dev_name = LanItems.GetDeviceName(itemId)
         ifcfg = LanItems.GetDeviceMap(itemId)
-        next if IsEmpty(dev_name)
+
+        next if dev_name.nil? || dev_name.empty?
+
         ifcfg = { "dev_name" => dev_name } if ifcfg == nil
         dev_type = LanItems.GetDeviceType(itemId)
-        if Builtins.contains(["tun", "tap"], dev_type)
+
+        if ["tun", "tap"].include? dev_type
           description = NetworkInterfaces.GetDevTypeDescription(dev_type, true)
         else
           description = BuildDescription(
-            "",
+            dev_type,
             "",
             ifcfg,
-            [Ops.get_map(LanItems.GetLanItem(itemId), "hwinfo", {})]
+            [LanItems.GetLanItem(itemId)["hwinfo"] || {}]
```

Comment:
What can I say. Yes it is. However, API is given and this patch  do not change this particular thing.